optuna/pruners/base.py

```diff
@@ -13,7 +13,7 @@ class BasePruner(object, metaclass=abc.ABCMeta):
     @abc.abstractmethod
     def prune(self, study, trial):
         # type: (Study, FrozenTrial) -> bool
-        """Judge whether the trial should be pruned at the given step.
+        """Judge whether the trial should be pruned at ``trial.last_step``.
```

Comment:
Actually, it's up to each pruner how to judge. How about something along "... should be pruned based on the reported values?"

Reply:
Hmm, maybe I don't get your point by mentioning "the reported values," so let me clarify my understanding.

Each concrete pruner class implements its `prune` method that judges whether to stop the given `trial` or not by basically comparing the reported value at `trial.last_step` with all the reported values.
However, the current docstring of this method does not mention the above thing, and is this why you suggest "based on reported values"?

Reply:
It's not a very strong opinion but my point was that we don't have to be that specific about the actual implementation in the documentation at the interface level.